### DISCUSSION TITLE
Convert old TED xml format to eForm xml Format

### DISCUSSION THREAD

**Discussion by FloB0** [OP-TED/eForms-SDK#798] [Q&A] [answered]
Hello TED Dev community,

I am working on transforming our old TED xml data into the new eForm format. Therefore, my aim is to use one of the APIs, but I have some issues figuring out how.

my current script looks like this:

```
import requests
import random

# API endpoint
url = 'https://enotices2.ted.europa.eu/esenders/api/v2/notice/notices/convert'
# url = 'https://enotices2.preview.ted.europa.eu/esenders/api/v2/notice/notices/convert'
# url = 'https://ted.europa.eu/api/v2/notice/notices/convert'


data = '<?xml version="1.0" encoding="UTF-8"?><TED_EXPORT xmlns:xlink="http://www.w3.org/1999/xlink" and so on'

proxy_server = []
proxy = random.choice(proxy_server)
proxies = {
    "http": proxy,
    "https": proxy,
    "ftp": proxy
}

headers = {
    "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7",
    "Accept-Encoding": "gzip, deflate",
    "Accept-Language": "en-US,en;q=0.9,de-DE;q=0.8,de;q=0.7",
    "Host": "httpbin.org",
    "Sec-Ch-Ua": "\"Not/A)Brand\";v=\"99\", \"Google Chrome\";v=\"115\", \"Chromium\";v=\"115\"",
    "Sec-Ch-Ua-Mobile": "?0",
    "Sec-Ch-Ua-Platform": "\"Windows\"",
    "Sec-Fetch-Dest": "document",
    "Sec-Fetch-Mode": "navigate",
    "Sec-Fetch-Site": "cross-site",
    "Sec-Fetch-User": "?1",
    "Upgrade-Insecure-Requests": "1",
    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/115.0.0.0 Safari/537.36",
    "X-Amzn-Trace-Id": "Root=1-64d35563-0edcadb001ed2e390e322e57"
}

response = requests.post(url, data=data, verify=False, headers=headers, proxies=proxies)

print(response.status_code)
print(response.text)

```

it uses three API endpoints:
url = 'https://enotices2.ted.europa.eu/esenders/api/v2/notice/notices/convert'
url = 'https://enotices2.preview.ted.europa.eu/esenders/api/v2/notice/notices/convert'
url = 'https://ted.europa.eu/api/v2/notice/notices/convert'

But I can't get any of them to work.

Note: I saw the xml_converter, but it is not a usable solution for my project unfortunately.

I hope you are well and thank you in advance for your help and ideas.
Flo

**Comment by pcolast** [comment, marked as answer]
Dear Flo,

In your request, the Host parameter is not set correctly.
The correct value is enotices2.ted.europa.eu or enotices2.preview.ted.europa.eu depending on the chosen environment.

To use the API, you need to have a valid API key for the given environment (Prod or Preview) and also Enotices2 account - please have a look at https://docs.ted.europa.eu/api/index.html#_api_keys
In your request you need to include header parameter
"Authorization": "Bearer value-of-your-api-key"

Have a nice day,
Stefan
